import "gopkg.in/yaml.v3" fails

### The following program `sample.go` triggers an unexpected result

```go
package main

import _ "gopkg.in/yaml.v3"

func main() {
    println("Hello")
}
```


### Expected result

```console
$ GO111MODULE=off go run ./sample.go
Hello
```


### Got

```console
$ yaegi ./sample.go
.go:3:8: import "gopkg.in/yaml.v3" error: /home/marc/go/src/gopkg.in/yaml.v3/decode.go:195:42: cannot use type *yaml.Node as type *unsafe2.dummy
```


### Yaegi Version

v0.10.0

### Additional Notes

_No response_

--- a/_test/issue-1315.go
+++ b/_test/issue-1315.go
@@ -1,0 +1,32 @@
+package main
+
+type Intf interface {
+	M()
+}
+
+type T struct {
+	s string
+}
+
+func (t *T) M() { println("in M") }
+
+func f(i interface{}) {
+	switch j := i.(type) {
+	case Intf:
+		j.M()
+	default:
+		println("default")
+	}
+}
+
+func main() {
+	var i Intf
+	var k interface{} = 1
+	i = &T{"hello"}
+	f(i)
+	f(k)
+}
+
+// Output:
+// in M
+// default

--- a/interp/run.go
+++ b/interp/run.go
@@ -2897,12 +2897,13 @@ func _case(n *node) {
 					if typ.cat == nilT && v.IsNil() {
 						return tnext
 					}
-					if typ.TypeOf().String() == t.String() {
+					rtyp := typ.TypeOf()
+					if rtyp != nil && rtyp.String() == t.String() && implementsInterface(v, typ) {
 						destValue(f).Set(v.Elem())
 						return tnext
 					}
 					ival := v.Interface()
-					if ival != nil && typ.TypeOf().String() == reflect.TypeOf(ival).String() {
+					if ival != nil && rtyp != nil && rtyp.String() == reflect.TypeOf(ival).String() {
 						destValue(f).Set(v.Elem())
 						return tnext
 					}
@@ -2963,6 +2964,22 @@ func _case(n *node) {
 			return fnext
 		}
 	}
+}
+
+func implementsInterface(v reflect.Value, t *itype) bool {
+	rt := v.Type()
+	if t.cat == valueT {
+		return rt.Implements(t.rtype)
+	}
+	vt := &itype{cat: valueT, rtype: rt}
+	if vt.methods().contains(t.methods()) {
+		return true
+	}
+	vi, ok := v.Interface().(valueInterface)
+	if !ok {
+		return false
+	}
+	return vi.node != nil && vi.node.typ.methods().contains(t.methods())
 }
 
 func appendSlice(n *node) {

